### PR TITLE
prevent consecutive navigation event

### LINKED ISSUE #971: Consecutive presses of back button on  screen cause a transition to  blank screen
**Describe the bug**
When I consective presses back button repeatedly on the following screen, transition to blank screen.
- `SearchScreen` 
- `TimetableItemDetailScreen`
- `StaffScreen`
- `SponsorsScreen`

**To Reproduce**
1. Go to Screens enumerated above
2. Consecutive click on back button

**Expected behavior**
No matter how many times I press back button, it does not to transition screen.

**Screenshots**
|SearchScreen|TimetableItemDetailScreen|StaffScreen|SponsorsScreen|
|---|---|---|---|
|<video src="https://github.com/DroidKaigi/conference-app-2023/assets/60963155/1be0bc0b-5eae-4938-9753-49ae6bd80c5b" width="300">|<video src="https://github.com/DroidKaigi/conference-app-2023/assets/60963155/57f480be-a201-497b-91ae-8f9cc6dba1cf" width="300">|<video src="https://github.com/DroidKaigi/conference-app-2023/assets/60963155/a3d9b1ec-104a-4f7b-9a86-1160910d77cd" width="300">|<video src="https://github.com/DroidKaigi/conference-app-2023/assets/60963155/958aca72-10f9-4a62-8ee9-a813cb515b07" width="300">|

## **Related issue**
https://github.com/DroidKaigi/conference-app-2023/issues/468

This issue has been corrected as follows
https://github.com/DroidKaigi/conference-app-2023/blob/99a14d95be8c98e7190862115e39935491928c1f/feature/sessions/src/main/java/io/github/droidkaigi/confsched2023/sessions/component/BookmarkTopArea.kt#L128-L134

### PR DESCRIPTION
**Pull request by ked4ma**
## Issue
- close #971

## Overview (Required)
- add handler method for click event
  - ignore events if other navigation has been started
- enable to back from contributor screen
  - now, if we back from contributor screen, empty screen is displayed (even if single tap)

## Links
- https://stackoverflow.com/questions/76386471/how-to-avoid-onclick-callback-being-called-multiple-times-jetpack-compose/76386604#76386604

## Movie
Target | Before | After
:--: | :--: | :--:
Searcj | <video src="https://github.com/DroidKaigi/conference-app-2023/assets/74723074/af82b34a-dcbd-4c32-9f15-42781d1f7c01" width="300" > | <video src="https://github.com/DroidKaigi/conference-app-2023/assets/74723074/4063450b-f75f-4b05-9e02-aae682a40352" width="300" >
TimetableItemDetail | <video src="https://github.com/DroidKaigi/conference-app-2023/assets/74723074/7fe299a1-c9cb-47b8-96b3-761f60836c03" width="300" > | <video src="https://github.com/DroidKaigi/conference-app-2023/assets/74723074/98085e57-0cd2-4e0d-a767-6bb533f385d2" width="300" >
Staff | <video src="https://github.com/DroidKaigi/conference-app-2023/assets/74723074/42c937a0-33bf-43a0-a3b6-dcedfce9eb83" width="300" > | <video src="https://github.com/DroidKaigi/conference-app-2023/assets/74723074/9d0f1735-5160-4f35-a00c-972f70c87083" width="300" >
Sponsor | <video src="https://github.com/DroidKaigi/conference-app-2023/assets/74723074/46d5488c-a5c3-48f1-ba6e-c0b59f918ca8" width="300" > | <video src="https://github.com/DroidKaigi/conference-app-2023/assets/74723074/fa9cef09-a9db-411c-a930-48de45bab36d" width="300" >
Contributor | <video src="https://github.com/DroidKaigi/conference-app-2023/assets/74723074/4163e392-6fbd-491e-9e10-ecb1d97cb5de" width="300" >| <video src="https://github.com/DroidKaigi/conference-app-2023/assets/74723074/2173401b-bf1e-4919-988f-a94aaa429125" width="300" >
Contributor(2) | <video src="https://github.com/DroidKaigi/conference-app-2023/assets/74723074/7acfaab1-9181-4291-9d3c-b3abc3524da9" width="300" >  | <video src="https://github.com/DroidKaigi/conference-app-2023/assets/74723074/1cc97e8d-e0b4-4741-b46a-2d544bda6743" width="300" >
 | <video src="" width="300" > | <video src="" width="300" >
Bookmark | <video src="https://github.com/DroidKaigi/conference-app-2023/assets/74723074/b82702f2-1c67-42d6-92f4-d678695801d2" width="300"> | <video src="https://github.com/DroidKaigi/conference-app-2023/assets/74723074/76c48ec9-39c4-4e96-8dac-9a536920343d" width="300" >

### CODE REVIEW

File: app-android/src/main/java/io/github/droidkaigi/confsched2023/KaigiApp.kt [hunk 1]
```diff
@@ -169,9 +171,15 @@ private fun NavGraphBuilder.mainScreen(
...
             // For KMP, we are not using navigation abstraction for contributors screen
             composable(contributorsScreenRoute) {
+                val lifecycleOwner = LocalLifecycleOwner.current
                 ContributorsScreen(
                     viewModel = hiltViewModel<ContributorsViewModel>(),
-                    onNavigationIconClick = navController::popBackStack,
+                    onNavigationIconClick = {
+                        handleOnClickIfNotNavigating(
+                            lifecycleOwner,
+                            mainNestedNavController::popBackStack,
```
Comment (ked4ma) [inline app-android/src/main/java/io/github/droidkaigi/confsched2023/KaigiApp.kt:180]: ContributorsScreen is navigated by `mainNestedNavControoler`.
So, changed to use same controller when call `popBackStack()`
By this, app became to pop back from contributors screen.

File: core/ui/src/androidMain/kotlin/io/github/droidkaigi/confsched2023/ui/OnClickHandler.kt [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ * Invoke "onClick" if current lifecycle isn't operating other navigation.
+ * @param owner a current LifecycleOwner
+ * @param onClick the method that want to invoke
+ */
+fun handleOnClickIfNotNavigating(owner: LifecycleOwner, onClick: () -> Unit) {
+    // https://stackoverflow.com/a/76386604/4339442
+    val currentState = owner.lifecycle.currentState
+    if (currentState.isAtLeast(Lifecycle.State.RESUMED)) {
+        onClick()
+    }
```
Comment (ked4ma) [inline core/ui/src/androidMain/kotlin/io/github/droidkaigi/confsched2023/ui/OnClickHandler.kt:16]: I add this method for reuse.
If you have better file name/package name/method name, please comment :bow:

File: app-android/src/main/java/io/github/droidkaigi/confsched2023/KaigiApp.kt [hunk 3]
```diff
@@ -17,7 +17,7 @@ import androidx.compose.runtime.Composable
 import androidx.compose.runtime.remember
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.platform.LocalContext
-import androidx.core.content.ContextCompat.startActivity
```
Comment (ked4ma) [inline app-android/src/main/java/io/github/droidkaigi/confsched2023/KaigiApp.kt:20]: that was already unused.

**Review by takahirom — APPROVED** [review]
Looks great!

**Pull request merged**